contracts/vaults/TreasuryVault/RibbonTreasuryVaultLite.sol

```diff
@@ -0,0 +1,919 @@
+    }
+
+    /************************************************
+     *  DEPOSIT & WITHDRAWALS
+     ***********************************************/
+
+    /**
+     * @notice Deposits the `asset` from msg.sender.
+     * @param amount is the amount of `asset` to deposit
+     */
+    function deposit(uint256 amount) external nonReentrant {
+        require(amount > 0, "!amount");
+
+        _depositFor(amount, msg.sender);
+
+        // An approve() by the msg.sender is required beforehand
+        IERC20(vaultParams.asset).safeTransferFrom(
+            msg.sender,
+            address(this),
+            amount
+        );
+    }
+
+    /**
+     * @notice Mints the vault shares to the creditor
+     * @param amount is the amount of `asset` deposited
+     * @param creditor is the address to receieve the deposit
+     */
+    function _depositFor(uint256 amount, address creditor) private {
+        uint256 currentRound = vaultState.round;
+        uint256 totalWithDepositedAmount = totalBalance().add(amount);
+
+        Vault.DepositReceipt memory depositReceipt = depositReceipts[creditor];
+
+        require(totalWithDepositedAmount <= vaultParams.cap, "Exceed cap");
+        require(
+            totalWithDepositedAmount >= vaultParams.minimumSupply,
+            "Insufficient balance"
+        );
+
+        emit Deposit(creditor, amount, currentRound);
+
+        // If we have an unprocessed pending deposit from the previous rounds, we have to process it.
+        uint256 unredeemedShares =
+            depositReceipt.getSharesFromReceipt(
+                currentRound,
+                roundPricePerShare[depositReceipt.round],
+                vaultParams.decimals
+            );
+
+        uint256 depositAmount = amount;
+
+        // If we have a pending deposit in the current round, we add on to the pending deposit
+        if (currentRound == depositReceipt.round) {
+            uint256 newAmount = uint256(depositReceipt.amount).add(amount);
+            depositAmount = newAmount;
+        }
+
+        ShareMath.assertUint104(depositAmount);
+
+        depositReceipts[creditor] = Vault.DepositReceipt({
+            round: uint16(currentRound),
+            amount: uint104(depositAmount),
+            unredeemedShares: uint128(unredeemedShares)
+        });
+
+        uint256 newTotalPending = uint256(vaultState.totalPending).add(amount);
+        ShareMath.assertUint128(newTotalPending);
+
+        vaultState.totalPending = uint128(newTotalPending);
+    }
+
+    /**
+     * @notice Initiates a withdrawal that can be processed once the round completes
+     * @param numShares is the number of shares to withdraw
+     */
+    function initiateWithdraw(uint256 numShares) external nonReentrant {
+        require(numShares > 0, "!numShares");
+
+        // We do a max redeem before initiating a withdrawal
+        // But we check if they must first have unredeemed shares
+        Vault.DepositReceipt storage depositReceipt =
```

Comment:
Could this be `memory` vs `storage`? (since we are not writing to storage with the `depositReceipt` variable)